src/include/rados/librados.hpp

```diff
@@ -1040,6 +1040,7 @@ namespace librados
     friend class ObjectWriteOperation;  // copy_from needs to see our IoCtxImpl
 
     IoCtxImpl *io_ctx_impl;
+    std::list<std::string> *cached_pool_names;
```

Comment:
this can be just cached_pool_names (not a pointer).. not need to put the list<> on the heap?